src/main/java/com/limechain/network/protocol/grandpa/GrandpaEngine.java

```diff
@@ -27,24 +26,46 @@ public class GrandpaEngine {
     private final SyncedState syncedState = SyncedState.getInstance();
 
     public void receiveRequest(byte[] msg, PeerId peerId, Stream stream) {
-        boolean connectedToPeer = connectionManager.isGrandpaConnected(peerId);
         GrandpaMessageType messageType = getGrandpaMessageType(msg);
 
         if (messageType == null) {
             log.log(Level.WARNING, String.format("Unknown grandpa message type \"%d\" from Peer %s", msg[0], peerId));
             return;
         }
 
+        if (stream.isInitiator()) {
+            handleInitiatorStreamMessage(msg, messageType, stream);
+        } else {
+            handleResponderStreamMessage(msg, messageType, stream);
+        }
```

Comment:
Since you've renamed `msg` to `message in the switch statement below, let's do that here as well in order to keep naming consistent.